lib/ret_web/controllers/page_controller.ex

```diff
@@ -792,4 +792,19 @@ defmodule RetWeb.PageController do
   defp with_script_tags(script), do: "<script>#{script}</script>"
 
   defp module_config(key), do: Application.get_env(:ret, __MODULE__)[key]
+
+  defp maybe_scrub_room_data(hub) do
+    if hub.entry_mode == :deny do
+      Map.merge(hub, %{
```

Comment:
I might just return a new struct here instead, we could easily add new fields and then forget to "scrub" them later.